plutus-core/plutus-ir/src/PlutusIR/Core/Instance/Pretty/Readable.hs

```diff
@@ -0,0 +1,145 @@
+{-# LANGUAGE ConstraintKinds       #-}
+{-# LANGUAGE LambdaCase            #-}
+{-# LANGUAGE MultiParamTypeClasses #-}
+{-# LANGUAGE OverloadedStrings     #-}
+{-# LANGUAGE TypeApplications      #-}
+{-# LANGUAGE TypeOperators         #-}
+{-# LANGUAGE UndecidableInstances  #-}
+{-# OPTIONS_GHC -Wno-orphans       #-}
+{-# OPTIONS_GHC -Wno-name-shadowing #-}
+{-# OPTIONS_GHC -Wno-incomplete-uni-patterns #-}
+module PlutusIR.Core.Instance.Pretty.Readable where
+
+import PlutusCore.Core.Instance.Pretty.Readable
+import PlutusCore.Default.Universe
+import PlutusCore.Pretty.ConfigName
+import PlutusCore.Pretty.PrettyConst
+import PlutusCore.Pretty.Readable
+import PlutusIR.Core.Type
+import PlutusPrelude
+import Prettyprinter
+
+
+prettyConfigReadable :: PrettyConfigReadable PrettyConfigName
+prettyConfigReadable = botPrettyConfigReadable defPrettyConfigName ShowKindsYes
+
+prettyReadable :: PrettyBy (PrettyConfigReadable PrettyConfigName) a => a -> Doc ann
+prettyReadable = prettyBy prettyConfigReadable
+
+viewApp :: Term tyname name uni fun ann
+        -> (Term tyname name uni fun ann, [Either (Type tyname uni ann) (Term tyname name uni fun ann)])
+viewApp t = go t []
+  where
+    go (Apply _ t s)  args = go t (Right s : args)
+    go (TyInst _ t a) args = go t (Left a : args)
+    go t args              = (t, args)
+
+vcatHard :: [Doc ann] -> Doc ann
```

Comment:
what does this do? haddock

Reply:
We have a module for prettyprinter extras: `Data.Prettyprinter.Custom`, if these are generically useful then put them there.